src/main/scala/com/databricks/spark/sql/perf/mllib/data/dataGeneration.scala

```diff
@@ -53,11 +54,60 @@ object DataGenerator {
       seed)
     sql.createDataFrame(rdd.map(Tuple1.apply)).toDF("features")
   }
+
+  def generateRatings(
+      sql: SQLContext,
+      numUsers: Int,
+      numProducts: Int,
+      numRatings: Long,
```

Comment:
How about taking numTestRatings so it's no longer hard-coded?  This will let you generate the right # of test examples below.